fs/accounting/stats.go

```diff
@@ -624,17 +624,28 @@ func (s *StatsInfo) RemoveTransfer(transfer *Transfer) {
 	s.mu.Unlock()
 }
 
-// PruneTransfers makes sure there aren't too many old transfers
-func (s *StatsInfo) PruneTransfers() {
+// pruneTransfers removes transfers that already finished
+//
+// Must be called with the lock held
+func (s *StatsInfo) pruneTransfers() {
+	for i, tr := range s.startedTransfers {
+		if tr.IsDone() {
+			s.removeTransfer(tr, i)
+			break
+		}
+	}
+}
+
+// PruneTransfers makes sure there aren't too many old transfers.
+// It purges transfers if number of startedTransfers exceeds `limit`.
+func (s *StatsInfo) PruneTransfers(limit int) {
```

Comment:
I think that naming / behaviour can be misleading here, how do ppl know what limit to put and how do they know that sth would be added to the number?

I'd suggest having two functions
* `PruneTransfers()` - the usual logic that uses `MaxCompletedTransfers`
* `PruneAllTransfers()` - remove all logic used by `clear`

Reply:
good idea, applied.